assets/admin/students/student-modal/index.js

```diff
@@ -10,86 +10,116 @@ import { __ } from '@wordpress/i18n';
 import InputControl from '../../../blocks/editor-components/input-control';
+import httpClient from './lib/http-client';
 
-let needsReload = false;
-
-const AddButton = (
-	<Button
-		className="sensei-student-modal__action--add"
-		variant="primary"
-		onClick={ () => {
-			needsReload = true;
-		} }
-	>
-		{ __( 'Add to Course', 'sensei-lms' ) }
-	</Button>
-);
-
-const RemoveButton = (
-	<Button
-		className="sensei-student-modal__action--remove"
-		onClick={ () => {
-			needsReload = true;
-		} }
-	>
-		{ __( 'Remove from Course', 'sensei-lms' ) }
-	</Button>
-);
-
-const ResetButton = (
-	<Button className="sensei-student-modal__action--remove">
-		{ __( 'Reset or Remove Progress', 'sensei-lms' ) }
-	</Button>
-);
+/**
+ * External dependencies
+ */
+import { useCallback, useRef, useState, useEffect } from 'react';
 
 const POSSIBLE_ACTIONS = {
 	add: {
 		description: __(
 			'Select the course(s) you would like to add students to:',
 			'sensei-lms'
 		),
-		button: AddButton,
+		buttonLabel: __( 'Add to Course', 'sensei-lms' ),
+		sendAction: ( students, courses ) =>
+			httpClient( {
+				restRoute: '/sensei-internal/v1/course-students/batch',
+				method: 'POST',
+				data: { student_ids: students, course_ids: courses },
+			} ),
 	},
 	remove: {
 		description: __(
 			'Select the course(s) you would like to remove students from:',
 			'sensei-lms'
 		),
-		button: RemoveButton,
+		buttonLabel: __( 'Remove from Course', 'sensei-lms' ),
+		sendAction: ( students, courses ) =>
+			httpClient( {
+				restRoute: '/sensei-internal/v1/course-students/batch',
+				method: 'DELETE',
+				data: { student_ids: students, course_ids: courses },
+			} ),
 	},
 	'reset-progress': {
 		description: __(
 			'Select the course(s) you would like to reset or remove progress for:',
 			'sensei-lms'
 		),
-		button: ResetButton,
+		buttonLabel: __( 'Reset or Remove Progress', 'sensei-lms' ),
+		sendAction: ( students, courses ) =>
+			httpClient( {
+				restRoute: '/sensei-internal/v1/course-progress/batch',
+				method: 'DELETE',
+				data: { student_ids: students, course_ids: courses },
+			} ),
 	},
 };
 
 /**
  * Questions modal content.
  *
  * @param {Object}   props
- * @param {Object}   props.action  Action that is being performed.
- * @param {Function} props.onClose Close callback.
+ * @param {Object}   props.action   Action that is being performed.
+ * @param {Function} props.onClose  Close callback.
+ * @param {Array}    props.students
  */
-export const StudentModal = ( { action, onClose } ) => {
-	const { description, button } = POSSIBLE_ACTIONS[ action ];
+export const StudentModal = ( { action, onClose, students } ) => {
+	const { description, buttonLabel, sendAction } = POSSIBLE_ACTIONS[ action ];
+	const selectedCourses = useRef( [] );
+	const [ isSending, setIsSending ] = useState( false );
+	const [ hasError, setError ] = useState( false );
+	const mounted = useRef( true );
+
+	useEffect( () => () => ( mounted.current = false ) );
```

Comment:
It is preventing a React warning due to the fact we are changing the component state after it is unmounted. 
The correct solution is to use [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) but it is breaking our jest setup, probably because our jest is old. I spend a lot of time trying to fix it without success.

Here an article explain the error.
https://medium.com/doctolib/react-stop-checking-if-your-component-is-mounted-3bb2568a4934

Reply:
Oh, I think I found a way to fix it better.

Reply:
I wouldn't mind seeing a comment right in the code that explains this, perhaps even with that link attached. Otherwise, I'd be likely to have the same question 6 months down the road.

Reply:
@donnapep yes, actually I am looking for better solutions to solve it. like https://github.com/closeio/use-abortable-effect